zict/buffer.py

```diff
@@ -153,4 +167,66 @@ def flush(self) -> None:
         flush(self.fast, self.slow)
 
     def close(self) -> None:
+        if self._executor is not None:
+            self._executor.shutdown(wait=True, cancel_futures=False)
+            self._executor = None
+
         close(self.fast, self.slow)
+
+    def _offload(self, func: Callable[[], T]) -> asyncio.Future[T]:
+        if self._executor is None:
+            self._executor = ThreadPoolExecutor(1)
+        loop = asyncio.get_running_loop()
+        return loop.run_in_executor(self._executor, func)
```

Comment:
This is a very early prototype. I still haven't decided if I'm going to go with a ThreadPoolExecutor or with a Queue.